tests/unit/amo/reducers/test_reviews.js

```diff
@@ -981,120 +987,160 @@ describe(__filename, () => {
-          reviewCount: 1,
           userId,
         }),
       );
       expect(prevState.byUserId[userId]).toBeDefined();
 
+      const _isReviewUpdate = sinon.stub().returns(true);
       const review = createInternalReview({
         ...fakeReview,
         user: {
           ...fakeReview.user,
           id: userId,
         },
       });
-      const state = addReviewToState({ state: prevState, review });
-      expect(state.byUserId[userId]).not.toBeDefined();
+      const state = _addReviewToState({
+        state: prevState,
+        review,
+        _isReviewUpdate,
+      });
+      expect(state.byUserId[userId]).toBeDefined();
     });
 
-    it('resets groupedRatings when adding a new rating', () => {
-      const addonId = 44231;
-      let state;
+    it('resets all related add-on reviews for new reviews', () => {
+      const addonSlug = 'some-slug';
 
+      let state;
       state = reviewsReducer(
         state,
-        setGroupedRatings({
-          addonId,
-          grouping: { 1: 0, 2: 0, 3: 0, 4: 0, 5: 0 },
+        _setAddonReviews({
+          addonSlug,
+          reviews: [{ ...fakeReview, id: 1 }],
         }),
       );
+      const _isReviewUpdate = sinon.stub().returns(false);
+      state = _addReviewToState({
+        state,
+        review: createInternalReview({
+          ...fakeReview,
+          addon: {
+            ...fakeReview.addon,
+            slug: addonSlug,
+          },
+          id: 2,
+        }),
+        _isReviewUpdate,
+      });
 
-      state = addReviewToState({
+      expect(state.byAddon[addonSlug]).toBeUndefined();
+    });
+
+    it('does not reset all related add-on reviews for updates', () => {
+      const addonSlug = 'some-slug';
+
+      let state;
+      state = reviewsReducer(
+        state,
+        _setAddonReviews({
+          addonSlug,
+          reviews: [{ ...fakeReview, id: 1 }],
+        }),
+      );
+      const _isReviewUpdate = sinon.stub().returns(true);
+      state = _addReviewToState({
         state,
         review: createInternalReview({
           ...fakeReview,
           addon: {
             ...fakeReview.addon,
-            id: addonId,
+            slug: addonSlug,
           },
+          id: 2,
         }),
+        _isReviewUpdate,
       });
 
-      expect(state.groupedRatings[addonId]).not.toBeDefined();
+      expect(state.byAddon[addonSlug]).toBeDefined();
     });
 
-    it('updates groupedRatings', () => {
+    it('resets groupedRatings when adding a new rating', () => {
       const addonId = 44231;
       let state;
 
-      const grouping1 = { 1: 0, 2: 0, 3: 0, 4: 0, 5: 0 };
-      const grouping2 = { 1: 1, 2: 1, 3: 1, 4: 1, 5: 1 };
-
       state = reviewsReducer(
         state,
-        setGroupedRatings({ addonId, grouping: grouping1 }),
+        setGroupedRatings({
+          addonId,
+          grouping: { 1: 0, 2: 0, 3: 0, 4: 0, 5: 0 },
+        }),
       );
-      state = reviewsReducer(
+
```

Comment:
The storage of grouped ratings is already covered by another test (below) so it would be redundant to re-test it.

https://github.com/mozilla/addons-frontend/blob/166c06d965446530bfcf168128ee011c4419ee04/tests/unit/amo/reducers/test_reviews.js#L1158-L1166